### ISSUE TITLE
Enhancement: [ban-types] Add `suggest` option to types

### ISSUE THREAD

**Issue by NotWoods** [typescript-eslint/typescript-eslint#6875]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/ban-types

### Description

Sometimes a banned type can support an autofix, but you'd prefer to pass an array of potential replacements rather than one autofix. 

Updating the `Types` option to:
```diff
{
  message: string;
  fixWith?: string;
+ suggest?: readonly string[];
}
```
would allow for replacements to be show as ESLint suggestions.

This would also allow the default `Object` and `{}` banned types to use suggestions.

### Fail

```typescript
function foo(x: {}) {}
```

### Pass

```typescript
function foo(x: object) {} // replaced by using ESLint suggestion
```
